### PR TITLE
:sparkles: feat: add passthrough for verbatim issue keys

### PR DESCRIPTION
**Pull request by djk01281**
- Closes #76

### Before vs After

- Previously the helper matched only `<prefix>/<number>` branches (e.g. `feature/123`), looked the prefix up in `rules`, and produced a GitHub reference like `#123` or `org/repo#123`. A branch carrying a full tracker key such as `feature/PROJ-123` matched nothing, so the commit stayed untagged.
- Now it also recognizes full issue keys (Jira, Linear, and similar, shaped `PROJECT-123`) for the projects you list, and copies them into the message verbatim. The existing `rules` behavior is unchanged and still wins when a branch matches both.

### Behavior by case

With this config:

```json
{
    "rules": { "feature": null, "qa": "org/repo" },
    "passthrough": ["PROJ"], // 명시 필요
    "protect": ["main"]
}
```

| Branch                       | Commit tag      | Resolver              | vs before |
| ---------------------------- | --------------- | --------------------- | --------- |
| `feature/123`                | `[#123]`        | rules                 | unchanged |
| `qa/45`                      | `[org/repo#45]` | rules                 | unchanged |
| `feature/PROJ-123`           | `[PROJ-123]`    | passthrough           | new       |
| `PROJ-123`                   | `[PROJ-123]`    | passthrough           | new       |
| `feature/PROJ-123-add-login` | `[PROJ-123]`    | passthrough           | new       |
| `feature/PROJ-123_wip`       | `[PROJ-123]`    | passthrough           | new       |
| `feature/OPS-9`              | untouched       | none (OPS not listed) | new       |
| `feature/12-ABC-34`          | `[#12]`         | rules wins            | unchanged |
| `feature/wip`                | untouched       | none                  | unchanged |
| `main`                       | commit blocked  | protect               | unchanged |

### passthrough

`passthrough` lists the project keys to copy verbatim; `rules` (GitHub prefixes) still take precedence when a branch matches both.

| Value          | Recognizes                     |
| -------------- | ------------------------------ |
| `["PROJ", …]`  | only the listed project keys   |
| omitted / `[]` | nothing (verbatim tagging off) |

Only listed projects are tagged, so unrelated `UPPERCASE-NUMBER` tokens (e.g. `UTF-8`, `SHA-1`) are never mistaken for issues.

### Branch recognition

`passthrough` recognizes a key shaped `PROJECT-NUMBER` - an uppercase project (`[A-Z][A-Z0-9]+`, **≥2 chars**), a hyphen, and a **1–7 digit** number - appearing anywhere in the branch. 

For a `passthrough` entry `PROJ`:

| Branch               | Tagged       | Why                                                             |
| -------------------- | ------------ | --------------------------------------------------------------- |
| `PROJ-123`           | `[PROJ-123]` | bare key                                                        |
| `feature/PROJ-123`   | `[PROJ-123]` | key after a path prefix                                         |
| `feature_PROJ-123`   | `[PROJ-123]` | characters before the key don't matter                          |
| `PROJ-123-add-login` | `[PROJ-123]` | description suffix (after the number) is ignored                |
| `PROJ-123_wip`       | `[PROJ-123]` | any suffix after the number is ignored                          |
| `PROJ-123-20260101`  | `[PROJ-123]` | a `-<number>` (date) suffix is ignored too                      |
| `PROJ-123/anything`  | `[PROJ-123]` | trailing path is fine                                           |
| `PROJ-12345678`      | not tagged   | the number has more than 7 digits                               |
| `PROJ/123`           | not tagged   | a key uses `-` (`PROJ-123`); the slash form is a `rules` prefix |
| `PROJ_123`           | not tagged   | project and number must be joined by `-`                        |
| `proj-123`           | not tagged   | the project part must be uppercase                              |
| `OPS-123`            | not tagged   | `OPS` is not listed                                             |

So name the branch with the key joined by a hyphen — `PROJ-123` or `feature/PROJ-123-short-desc`. `PROJ_123` (underscore) and `PROJ/123` (slash) are not keys; a suffix *after* the number (`-desc`, `_wip`, `-20260101`, `/path`) is fine.

### Configuration example

```json
{
    "$schema": "./node_modules/@naverpay/commithelper-go/schema.json",
    "rules": { "feature": null, "qa": "org/repo" },
    "passthrough": ["PROJ"],
    "protect": ["main", "release/*"],
    "template": "{{.Message}}\n\nRef. [{{.Prefix}}]"
}
```

`$schema` is optional; it gives editor autocompletion and validation and resolves from the installed package (offline, no CDN).

### Fixes included

**Idempotency (`isAlreadyTagged` → `alreadyHasRef`).** The old check treated a message as tagged only when it *started* with a bracketed reference matching `^\[.*?#\d+\]`. That was tied to the default format and the `#number` shape, so a custom template placing the reference in the body went undetected and was re-added on every amend. It is replaced by `alreadyHasRef`, which checks whether the branch's resolved reference is already present anywhere in the message **as a whole token** (so `#123` is not found in `#1234`, and `PROJ-1871` is not found in `MY-PROJ-1871`). This is template-independent and works for verbatim keys too.

On branch `feature/123` (resolved reference `#123`):

| Incoming message                   | Old `isAlreadyTagged` | New `alreadyHasRef` |
| ---------------------------------- | --------------------- | ------------------- |
| `[#123] fix`                       | skip                  | skip                |
| `fix` with `Ref. [#123]` in body   | re-tag (bug)          | skip                |
| `[#999] fix` (different reference) | skip                  | adds `[#123]`       |
| `fix`                              | tag                   | tag                 |

The third row is the one intentional behavior change: the old check skipped when the message started with *any* bracketed reference; the new check skips only when *this branch's own* reference is present.

**Protected branch ordering.** Branch protection now runs before the already-tagged check, so a message that already contains a tag can no longer slip past `protect`.

### Tests

`resolveKey` is covered by a table that mirrors **Jigit's rules** (`TestResolveKey_JigitParity`) - every branch Jigit's docs list as recognized/ignored is asserted here - plus edge cases (8-digit numbers, `_`/date suffixes, longer-key boundaries, unlisted projects). `alreadyHasRef` and `processMessage` cover the idempotency and protect behavior above.

### CODE REVIEW

**Review by yujeong-jeon — DISMISSED** [review]
LGTM~!

File: packages/commithelper-go/main.go [hunk 1]
```diff
@@ -26,77 +27,156 @@ type TemplateData struct {
...
+	if len(passthrough) == 0 {
+		return nil
+	}
+	allowed := make(map[string]bool, len(passthrough))
+	for _, k := range passthrough {
+		allowed[k] = true
+	}
+	for _, m := range keyPattern.FindAllStringSubmatch(branch, -1) {
+		if len(m[2]) <= 7 && allowed[m[1]] {
+			return &TemplateData{Prefix: m[0]}
```
Comment (kyungmi) [inline packages/commithelper-go/main.go]: **[nit]** passthrough 경로에서는 `Prefix` 만 채워서, 커스텀 `template` 이 `{{.Number}}`(또는 `{{.Repo}}`)를 참조하면 빈 값이 렌더될 수 있어요.

- 예: `"template": "{{.Message}}\n\nRef. [#{{.Number}}]"` 설정 시 `feature/PROJ-123` 브랜치에서는 `Ref. [#]` 로 나옵니다. (rules 경로는 `Number` 가 채워져 정상)
- 기본 template(`[{{.Prefix}}] {{.Message}}`)에는 영향이 없어 우선순위는 낮습니다. `Number: m[2]` 정도를 함께 채워두면 기존 template 사용자와의 호환이 깔끔해질 것 같아요.

File: packages/commithelper-go/main.go [hunk 2]
```diff
@@ -26,77 +27,156 @@ type TemplateData struct {
...
 	}
 
 	input := os.Args[1]
+	isFile := false
 	var commitMessage string
 
 	if _, err := os.Stat(input); err == nil {
-		// Input is a file path
+		isFile = true
 		commitMessageBytes, err := ioutil.ReadFile(input)
```
Comment (kyungmi) [inline packages/commithelper-go/main.go:49]: **[nit]** `io/ioutil` 은 Go 1.16 부터 deprecated 라(현재 go.mod `1.24.4`), 이 리팩터링 겸 `os`/`io` 로 정리해도 좋을 것 같아요.

- `ioutil.ReadFile`/`WriteFile` → `os.ReadFile`/`os.WriteFile`, `ioutil.ReadAll` → `io.ReadAll`
- 이 PR 스코프(passthrough)와는 무관해서 별도 정리로 빼도 무방합니다.

**Review by kyungmi — DISMISSED** [review]
main() 에서 순수 코어 processMessage 를 분리하고 resolve/resolvePrefix/resolveKey/alreadyHasRef 로 쪼갠 덕에 테이블 테스트가 촘촘하게 붙은 구조가 좋습니다.

- passthrough 를 allowlist + 프로젝트 키(≥2자)·숫자(1~7자리)로 좁혀 UTF-8/SHA-1 같은 토큰 오탐을 막고, alreadyHasRef 의 토큰 경계 검사로 `[PROJ-123]` 재태깅까지 멱등 처리한 점이 견고합니다.
- rules 우선 → passthrough 순서와 Jigit parity 테스트까지 있어 동작 확신이 섭니다.
- [nit] 2건만 남깁니다: 커스텀 template + passthrough 조합에서 `{{.Number}}` 빈 값 / `ioutil` deprecated 정리. 둘 다 반영은 선택입니다.

File: packages/commithelper-go/main.go (hunk 1 above)
Comment (djk01281) [reply packages/commithelper-go/main.go]: f5bd805 수정했습니다! 한번만 더 부탁드립니닷 !

File: packages/commithelper-go/main.go (hunk 2 above)
Comment (djk01281) [reply packages/commithelper-go/main.go:49]: 별도로 나중에 수정해보겠습니다! 감사합니다!

**Review by keemhyunseok — APPROVED** [review]
감사합니다
